content/articles/an-introduction-to-blockchain-architecture/index.md

```diff
@@ -0,0 +1,113 @@
+
+### Distributed P2P Network
+Here is an image view of what a distributed peer-to-peer network looks like:
+
+![A Distributed Network - Image Credit(Wikipedia)](/engineering-education/an-introduction-to-blockchain-architecture/p2p.png)
+
+
+A distributed network consists of computers commonly referred to as nodes in the blockchain world, and they are interconnected, but they are not all connected simultaneously.
+
+When a block is mined in a blockchain, copies are made available across all interconnected computers on the network. Still, the data are all encrypted through cryptographic keys.
+
+In a distributed P2P network, once a new block is added to the blockchain, the block is added to all computers across the network but not simultaneously. As a result, some computers on the network will receive a block update faster than others, resulting in a problem, but we will get to this under consensus protocol.
+
+The realm of Distributed P2P ensures that not one single central party have the blockchain but rather thousands of systems in the network, and this protects against any form of attack, such that if a block is corrupted, the network can look across the distributed P2P and copy the last active and correct blockchain from the network to replace the invalid blockchain on the network.
+
+I find the function of the distributed P2P compelling and astonishing.
+
+### Hash Cryptography
+In blockchain, there is the need for unique identification of each block, like a fingerprint. The way to ensure such is through an algorithm developed by the National Security Agency(NSA), the Secure Hash Algorithm 256 (SHA256).
+
+[Here](https://emn178.github.io/online-tools/sha256.html) is a link to test run the SHA256 algorithm. From the web application online, you can generate a hash of whatever text you input. Here’s what mine looks like:
+
+
+![Hash Example](/engineering-education/an-introduction-to-blockchain-architecture/hash.png)
+
+
+The text input of hello world produces the hash. If I change a single text or add a single space, this changes the hash completely - you can experiment on this.
+
+In the blockchain, the data in the block + the previous hash and other properties are used to generate the hash for the block, which serves as a unique fingerprint for the block.
+
+### Mining and Proof of Work
+Mining is a broad topic that deserves its write-up. In this article, I’ll give an illustration of how mining works.
+
+For a block to be mined and added to the blockchain after being verified, there is usually a target for miners to attain, and this is achieved by changing a field in the block until the block's hash falls below the target of the pool of hashes.
+
+We’ve established that one cannot change the data in the field, so how do we make the hash change until it falls in the target zone? Well, two fields change, and they cause the hash to change. These fields are Timestamp and Nonce. For this article, I’ll limit my focus to just the Nonce field.
```

Comment:
You should focus on timestamp as well. It is equally important to explain the process of mining.

Reply:
I omitted the topic of time stamp intentionally as it’s a new field to the block I aim to talk extensively about in the next article I aim to write. This a basic introductory article.